tests/gflownet/envs/common.py

```diff
@@ -392,7 +392,7 @@ def test__trajectories_are_reversible(env):
     while not env.done:
         state, action, valid = env.step_random(backward=False)
         if valid:
-            states_trajectory_fw.append(state)
+            states_trajectory_fw.append(copy(state))
```

Comment:
What's going on with this copy? Maybe worth a comment (I assume it is so the state does not get changed in memory).

If so, are you sure this is working as intended, and should not instead be a deepcopy?

Reply:
Uhm I'm never 100 % sure with these things, but I think so. We want to store the different states of a trajectory in a list. I think most environments copy the state before applying the step, which is why this has always worked as expected. But it should not be mandatory for the step method to copy the state. If that doesn't happen, then this test method should do the copy to save what we want without risking it will be changed.

Now, about deepcopy. The `copy()` method here refers to [this method](https://github.com/alexhernandezgarcia/gflownet/blob/misc/gflownet/utils/common.py#L272) from the common utils. We're copying a state, so that should be fine, no? No need for a deep copy, I think.